src/FffGcodeWriter.cpp

```diff
@@ -1660,57 +1732,93 @@ bool FffGcodeWriter::addSupportInfillToGCode(const SliceDataStorage& storage, La
+    {
+        infill_outline_list.push_back(support_layer.support_infill_part_list[i].outline);
+    }
     PathOrderOptimizer island_order_optimizer(gcode_layer.getLastPosition());
-    for(unsigned int n=0; n<support_islands.size(); n++)
+    for (unsigned int i = 0; i < infill_outline_list.size(); ++i)
     {
-        island_order_optimizer.addPolygon(support_islands[n][0]);
+        island_order_optimizer.addPolygon(infill_outline_list[i][0]);
     }
     island_order_optimizer.optimize();
 
-    for(unsigned int n=0; n<support_islands.size(); n++)
+    // process each support infill islands
+    for (int support_infill_part_idx : island_order_optimizer.polyOrder)
     {
-        PolygonsPart& island = support_islands[island_order_optimizer.polyOrder[n]];
+        const SupportInfillPart& support_infill_part = support_layer.support_infill_part_list[support_infill_part_idx];
 
-        int support_infill_overlap = 0; // support infill should not be expanded outward
-        
-        int offset_from_outline = 0;
+        // add outline (boundary)
+        const Polygons& outline = support_infill_part.outline;
         if (support_pattern == EFillMethod::GRID || support_pattern == EFillMethod::TRIANGLES || support_pattern == EFillMethod::CONCENTRIC)
         {
-            Polygons boundary = island.offset(-support_line_width / 2);
+            Polygons boundary = outline.offset(-support_line_width / 2);
             if (boundary.size() > 0)
             {
                 setExtruder_addPrime(storage, gcode_layer, layer_nr, infill_extruder_nr_here); // only switch extruder if we're sure we're going to switch
                 gcode_layer.setIsInside(false); // going to print stuff outside print object, i.e. support
                 gcode_layer.addPolygonsByOptimizer(boundary, &gcode_layer.configs_storage.support_infill_config);
             }
-            offset_from_outline = -support_line_width;
-            support_infill_overlap = infill_extr_here.getSettingInMicrons("infill_overlap_mm"); // support lines area should be expanded outward to overlap with the boundary polygon
         }
 
-        int extra_infill_shift = 0;
-        bool use_endpieces = true;
-        Polygons* perimeter_gaps = nullptr;
-        double fill_angle = 0;
-        Infill infill_comp(support_pattern, island, offset_from_outline, support_line_width, support_line_distance, support_infill_overlap, fill_angle, z, extra_infill_shift, perimeter_gaps, infill_extr.getSettingBoolean("support_connect_zigzags"), use_endpieces);
-        Polygons support_polygons;
-        Polygons support_lines;
-        infill_comp.generate(support_polygons, support_lines);
-        if (support_lines.size() > 0 || support_polygons.size() > 0)
+        // process sub-areas in this support infill area with different densities
+        for (uint32_t density_idx = 0; density_idx < support_infill_part.gradual_infill_areas.size(); ++density_idx)
```

Comment:
Again I find this type oddly specific.
I would have done just `unsigned int` here.
Maybe that's just a personal preference, because I think those two types evaluate to the same on every system, though I'm not sure.

Actually, because the density has a maximum specified in the frontend `unsigned char` should even do! :D 

Still I'd vote for `unsigned int`